mmds/src/data_store.rs

```diff
@@ -31,19 +42,44 @@ impl Mmds {
     /// When a PATCH request is made on an uninitialized Mmds structure the status
     /// code should be 404 (Not Found) otherwise the returned status code should be
     /// 204 (No Content).
-    pub fn is_initialized(&self) -> bool {
-        self.is_initialized
+    pub fn is_initialized(&self) -> Result<(), Error> {
```

Comment:
Does this function need to be public? Also, since it doesn't return the `is_initialized` field any more, we should rename it to something more appropriate (maybe `check_data_store_initialized` although it's a rather long name and I am also not a big fan of prefixing function names with "check"; any other idea is welcome).
We also need to update the comment above the function as it doesn't seem relevant anymore.

Reply:
Oh, you are right! It doesn't need to be a public function.